test/unit/kubernetes-deploy/runner_task_test.rb

```diff
@@ -1,50 +1,27 @@
 # frozen_string_literal: true
 require 'test_helper'
+require 'kubernetes-deploy/runner_task'
 
 class RunnerTaskUnitTest < KubernetesDeploy::TestCase
-  def test_missing_namespace
-    assert_raises(KubernetesDeploy::RunnerTask::FatalTaskRunError,
-      message: "Configuration invalid: Namespace was not found") do
-      task_runner = KubernetesDeploy::RunnerTask.new(
-        context: KubeclientHelper::MINIKUBE_CONTEXT,
-        namespace: "missing",
-      )
-
-      task_runner.run(
-        task_template: 'hello-cloud-template-runner',
-        entrypoint: nil,
-        args: 'a'
-      )
-    end
-  end
-
-  def test_missing_arguments
-    assert_raises(KubernetesDeploy::RunnerTask::FatalTaskRunError) do
-      task_runner = KubernetesDeploy::RunnerTask.new(
-        context: KubeclientHelper::MINIKUBE_CONTEXT,
-        namespace: @namespace,
-      )
-
-      task_runner.run(
-        task_template: 'hello-cloud-template-runner',
-        entrypoint: nil,
-        args: nil
-      )
-    end
+  def setup
+    Kubeclient::Client.any_instance.stubs(:discover)
```

Comment:
That would seem better, but the problem is that our `build_kubeclient` method eagerly call `discover`, which means that if the kubeclient instance so much as gets created, discover will get called. Tbh the errors that happen if the test does end up calling a kubeclient method is super ugly; since discover was stubbed out, none of the `get_` (etc.) methods are defined, so it's a NoMethodError. I'm not sure what to do about that though.